src/lib/link.spec.ts

```diff
@@ -484,3 +485,85 @@ test.serial(
     t.deepEqual(postAcks[1], acks[2]);
   }
 );
+
+test.serial.only('timeout expired packets', async (t) => {
+  const [nodeA, nodeB] = await setup();
+
+  const link = await Link.createWithNewConnections(nodeA, nodeB);
+  const channels = await link.createChannel(
+    'A',
+    ics20.srcPortId,
+    ics20.destPortId,
+    ics20.ordering,
+    ics20.version
+  );
+
+  // no packets here
+  const noPackets = await link.endA.querySentPackets();
+  t.is(noPackets.length, 0);
+
+  // some basic setup for the transfers
+  const recipient = randomAddress(wasmd.prefix);
+  const latestHeight = (await nodeB.latestHeader()).height;
+  const timeoutDestHeight = latestHeight + 2;
+  const submitDestHeight = latestHeight + 500; // valid for 500 blocks
+  const amounts = [1000, 2222, 3456];
+  const timeoutHeights = [
+    submitDestHeight,
+    timeoutDestHeight,
+    submitDestHeight,
+  ];
+
+  // let's make 3 transfer tx at different heights
+  const txHeights = [];
+  for (let i = 0; i < amounts.length; ++i) {
+    const token = { amount: amounts[i].toString(), denom: simapp.denomFee };
+    const { height } = await nodeA.transferTokens(
+      channels.src.portId,
+      channels.src.channelId,
+      token,
+      recipient,
+      timeoutHeights[i]
```

Comment:
It would be good to make:

1. expired via height
2. non-expired
3. expired via time